numba/core/interpreter.py

```diff
@@ -3164,3 +3296,31 @@ def op_LOAD_METHOD(self, *args, **kws):
 
     def op_CALL_METHOD(self, *args, **kws):
         self.op_CALL_FUNCTION(*args, **kws)
+
+    if PYVERSION in ((3, 12), ):
+        def op_CALL_INTRINSIC_1(self, inst, intrinsic, **kwargs):
+            if intrinsic == "INTRINSIC_STOPITERATION_ERROR":
+                stmt = ir.StaticRaise(INTRINSIC_STOPITERATION_ERROR, (),
+                                      self.loc)
+                self.current_block.append(stmt)
+                return
+            elif intrinsic == "UNARY_POSITIVE":
+                self.op_UNARY_POSITIVE(inst, **kwargs)
+                return
+            elif intrinsic == "INTRINSIC_LIST_TO_TUPLE":
+                self.op_LIST_TO_TUPLE(inst, **kwargs)
+                return
+            raise NotImplementedError(intrinsic)
+    elif PYVERSION in ((3, 8), (3, 9), (3, 10), (3, 11)):
+        pass
+    else:
+        raise NotImplementedError(PYVERSION)
+
+
+if PYVERSION in ((3, 12), ):
+    class INTRINSIC_STOPITERATION_ERROR(AssertionError):
```

Comment:
Why does this extend from `AssertionError`? `StopIteration` MRO has `Exception` as its base?

Reply:
Further thought on this, should the `ir.StaticRaise` above just raise a `StopIteration`, i.e. this class isn't needed?